### DISCUSSION TITLE
Writing component to custom folder

### DISCUSSION THREAD

**Discussion by spamdwich** [shadcn-ui/ui#4562] [Q&A]
|── components/
│   ├── Button/
│   │   ├── index.tsx
│   │   ├── Button.css
│   │   └── Button.test.tsx

This is how the directory structure of my project looks like. I used shadcn and it adds the component file inside the "ui" alias that we have. Is there any flag in cli using which I can add the file to a custom path. I can import the components using relative path. Is there any way right now?

**Comment by spamdwich** [comment]
Ended up finding a solution here - [documentation](https://ui.shadcn.com/docs/cli#options-1)

**Reply by kerryj89** [reply]
How were you able to do this with I'm assuming the `path` option?
I am also after the folder-per-component structure like yours.
It doesn't seem to work for most of us https://github.com/shadcn-ui/ui/issues/848

--

Actually, a clue from https://github.com/shadcn-ui/ui/issues/4680#issuecomment-2377646092 shows that it used to work

**Comment by L3on06** [comment]
To customize the installation path for ShadCN/UI components, follow these steps:

1. Navigate to the components.json file.

- If this file does not exist at the root level, run the command:

```
npx shadcn@latest init
```

2. In the components.json file, update the following properties:

- ui: Specify the path where the components should be installed.
- components: Define the path for the components' installation.

Example:

```
{
  "ui": "@/shared/components/ui",
  "components": "@/shared/components"
}
```

**Comment by thekenna** [comment]
`npx shadcnf@latest add button` 
https://github.com/someError/shadcn-folder-component
https://www.npmjs.com/package/shadcnf

**Comment by aperkaz** [comment]
A bit late to the thread, but the CLI flag `add -p` is all you need.
